collect_app/src/main/java/org/odk/collect/android/formmanagement/BlankFormsListViewModel.java

```diff
@@ -0,0 +1,78 @@
+package org.odk.collect.android.formmanagement;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import androidx.lifecycle.ViewModelProvider;
+
+import org.odk.collect.android.forms.FormRepository;
+import org.odk.collect.android.forms.MediaFileRepository;
+import org.odk.collect.android.openrosa.api.FormAPI;
+import org.odk.collect.android.openrosa.api.FormAPIError;
+import org.odk.collect.android.utilities.MultiFormDownloader;
+import org.odk.collect.async.Scheduler;
+
+import javax.inject.Inject;
+
+public class BlankFormsListViewModel extends ViewModel {
+
+    private final Scheduler scheduler;
+    private final FormRepository formRepository;
+    private final MediaFileRepository mediaFileRepository;
+    private final FormAPI formAPI;
+    private final MultiFormDownloader multiFormDownloader;
+
+    private final MutableLiveData<Boolean> syncing = new MutableLiveData<>(false);
+
+    public BlankFormsListViewModel(Scheduler scheduler, FormRepository formRepository, MediaFileRepository mediaFileRepository, FormAPI formAPI, MultiFormDownloader multiFormDownloader) {
+        this.scheduler = scheduler;
+        this.formRepository = formRepository;
+        this.mediaFileRepository = mediaFileRepository;
+        this.formAPI = formAPI;
+        this.multiFormDownloader = multiFormDownloader;
+    }
+
+    public LiveData<Boolean> isSyncing() {
+        return syncing;
+    }
+
+    public void syncWithServer() {
+        syncing.setValue(true);
+
+        scheduler.scheduleInBackground(() -> {
+            try {
+                ServerFormListSynchronizer synchronizer = new ServerFormListSynchronizer(formRepository, mediaFileRepository, formAPI, multiFormDownloader);
+                synchronizer.synchronize();
+            } catch (FormAPIError ignored) {
+                // Ignored
+            }
+
+            return null;
+        }, ignored -> syncing.setValue(false));
+    }
+
+    public static class Factory implements ViewModelProvider.Factory {
+
+        private final Scheduler scheduler;
+        private final FormRepository formRepository;
+        private final MediaFileRepository mediaFileRepository;
+        private final FormAPI formAPI;
+        private final MultiFormDownloader multiFormDownloader;
+
+        @Inject
```

Comment:
I had completely forgotten we could do this. I'm not usually a huge fan of `@Injects` scattered around the place randomly building the dependency graph for the app - as much as it's terse give me a 500 line module definition any day. However, in the case of `ViewModel` where we have to deal with Factory classes like this I think just being able to `@Inject` the constructor and then injecting it in the Activity cuts down on a lot of explicit dependencies at the `Activity` level that only get used for `ViewModel` construction.

Probably a nice trick to use elsewhere.